src/value.rs

```diff
@@ -388,6 +389,138 @@ impl<'a> Iterator for PaddedBitsIter<'a> {
     }
 }
 
+trait Padding {
+    fn read_left_padding<I: Iterator<Item = bool>>(
+        bits: &mut I,
+        ty_l: &Final,
+        ty_r: &Final,
+    ) -> Result<(), EarlyEndOfStreamError>;
+
+    fn read_right_padding<I: Iterator<Item = bool>>(
+        bits: &mut I,
+        ty_l: &Final,
+        ty_r: &Final,
+    ) -> Result<(), EarlyEndOfStreamError>;
+}
+
+struct CompactEncoding;
+struct PaddedEncoding;
```

Comment:
In 34e1243f36022f88376fcaf810145d539a6aa669:

These should be empty enums. Otherwise they will have constructors.

Reply:
Thanks for pointing this out. I pushed a fix.